### PR TITLE
docs: update some docs for v0.4

### LINKED ISSUE #485: Update docs for feat: support desc [table] <table_name>
A document change request is generated from
https://github.com/GreptimeTeam/greptimedb/pull/1944

### PR DESCRIPTION
**Pull request by killme2008**
* Fixed #485 , #576 
* Adds SQL compatibility

# Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.

### CODE REVIEW

File: docs/en/v0.4/reference/sql/compatibility.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+Sure, here's the translation:
```
Comment (nicecui) [inline docs/en/v0.4/reference/sql/compatibility.md:1]:
```suggestion

```

File: docs/zh/v0.4/reference/sql/compatibility.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
 
-TODO
+GreptimeDB 支持的 SQL 是 ANSI SQL 的子集，并且拥有一些特有的扩展。一些主要的不兼容和扩展描述如下：
+
+1. 建表：
+  * 支持特有的 `TIME INDEX` 约束，详细请参考[数据模型](/user-guide/concepts/data-model)和 [CREATE](./create.md) 建表语法一节。
+  * 目前仅支持 `PRIMARY KEY` 约束，不支持其他类型的约束，也不支持外键。
+  * GreptimeDB 是原生的分布式数据库，因此分布式表的建表语句支持分区规则，也请参考[CREATE](./create.md) 建表语法一节。
+2. 插入： 与 ANSI SQL 语法一致，但是强制要求提供 `TIME INDEX` 列值（或默认值）。
+3. 更新：不支持 `UPDATE` 语法，但是在 `INSERT` 的时候，如果主键和 `TIME INDEX` 对应的列值一样，那么后续插入的行将覆盖以前写入的行，从而变相实现更新。
```
Comment (nicecui) [inline docs/zh/v0.4/reference/sql/compatibility.md:10]:
```suggestion
3. 更新数据：不支持 `UPDATE` 语法，但是在 `INSERT` 的时候，如果 `TIME INDEX` 对应的值在表中已存在，那么后续插入的行数据将覆盖以前写入的行数据，从而变相实现更新。
```

File: docs/zh/v0.4/reference/sql/compatibility.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
 # ANSI Compatibility
 
-TODO
+GreptimeDB 支持的 SQL 是 ANSI SQL 的子集，并且拥有一些特有的扩展。一些主要的不兼容和扩展描述如下：
+
+1. 建表：
+  * 支持特有的 `TIME INDEX` 约束，详细请参考[数据模型](/user-guide/concepts/data-model)和 [CREATE](./create.md) 建表语法一节。
+  * 目前仅支持 `PRIMARY KEY` 约束，不支持其他类型的约束，也不支持外键。
+  * GreptimeDB 是原生的分布式数据库，因此分布式表的建表语句支持分区规则，也请参考[CREATE](./create.md) 建表语法一节。
+2. 插入： 与 ANSI SQL 语法一致，但是强制要求提供 `TIME INDEX` 列值（或默认值）。
```
Comment (nicecui) [inline docs/zh/v0.4/reference/sql/compatibility.md:9]:
```suggestion
2. 插入新数据： 与 ANSI SQL 语法一致，但是强制要求提供 `TIME INDEX` 列值（或默认值）。
```

File: docs/zh/v0.4/reference/sql/compatibility.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+3. 更新：不支持 `UPDATE` 语法，但是在 `INSERT` 的时候，如果主键和 `TIME INDEX` 对应的列值一样，那么后续插入的行将覆盖以前写入的行，从而变相实现更新。
+4. 查询：查询语法兼容 ANSI SQL，存在部分功能差异和缺失
+  * 不支持视图
+  * TQL 语法扩展：通过 TQL 子命令支持在 SQL 里执行 PromQL，详细请参考 [TQL](./tql.md) 一节。
```
Comment (nicecui) [inline docs/zh/v0.4/reference/sql/compatibility.md:13]:
```suggestion
  * TQL 语法扩展：TQL 子命令支持在 SQL 中执行 PromQL，详细请参考 [TQL](./tql.md) 一节。
```

File: docs/zh/v0.4/reference/sql/compatibility.md [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+5. 删除：删除语法与 ANSI SQL 基本一致。
```
Comment (nicecui) [inline docs/zh/v0.4/reference/sql/compatibility.md:14]:
```suggestion
5. 删除数据：语法与 ANSI SQL 基本一致。
```

File: docs/zh/v0.4/reference/sql/compatibility.md [hunk 6]
```diff
@@ -1,3 +1,17 @@
...
+6. 他项：
+  * 标识符，如表名，列名等，约束与 ANSI SQL 类似，大小写敏感，遇到特殊字符或者大写需要用双引号括起来。
+  * GreptimeDB 做了个优化，针对不同方言，比如你用 MySQL 客户端或者 PostgreSQL 客户端连接上来， 允许使用特定 SQL 方言的标识符规则，比如 MySQL 可以用反引号 ```，而 PostgreSQL 还是标准的双引号 `"`。
```
Comment (nicecui) [inline docs/zh/v0.4/reference/sql/compatibility.md:17]:
```suggestion
  * GreptimeDB 针对不同方言做了优化，比如用 MySQL 客户端或者 PostgreSQL 客户端连接到数据库时， 允许使用特定 SQL 方言的标识符规则，比如在 MySQL 中可以用反引号 ```，而在 PostgreSQL 中还是标准的双引号 `"`。
```

**Review by nicecui — COMMENTED** [review]
👍

File: docs/zh/v0.4/reference/sql/compatibility.md (hunk 2 above)
Comment (killme2008) [reply docs/zh/v0.4/reference/sql/compatibility.md]: 这个不对，time index 相同不会覆盖，是需要主键 + time index 都相同才会

File: docs/en/v0.4/reference/sql/compatibility.md [hunk 7]
```diff
@@ -1,3 +1,18 @@
...
 
-TODO
+GreptimeDB supports a subset of ANSI SQL and has some unique extensions. Some major incompatibilities and extensions are described below:
+
+1. Table creation:
+   * Supports the unique `TIME INDEX` constraint. Please refer to the [Data Model](/user-guide/concepts/data-model) and the [CREATE](./create.md) table creation syntax for details.
+   * Currently only supports `PRIMARY KEY` constraints and does not support other types of constraints or foreign keys.
+   * GreptimeDB is a native distributed database, so the table creation syntax for distributed tables supports partitioning rules. Please also refer to the [CREATE](./create.md) table creation syntax.
+2. Insertion: Consistent with ANSI SQL syntax, but requires the `TIME INDEX` column value (or default value) to be provided.
+3. Update: Does not support `UPDATE` syntax, but if the primary key and `TIME INDEX` corresponding column values are the same during `INSERT`, subsequent inserted rows will overwrite previously written rows, effectively achieving an update.
```
Comment (nicecui) [inline docs/en/v0.4/reference/sql/compatibility.md:11]:
```suggestion
3. Update data: Does not support `UPDATE` syntax, but if the primary key and `TIME INDEX` corresponding column values are the same during `INSERT`, subsequent inserted rows will overwrite previously written rows, effectively achieving an update.
```

File: docs/en/v0.4/reference/sql/compatibility.md [hunk 8]
```diff
@@ -1,3 +1,18 @@
...
+4. Query: Query syntax is compatible with ANSI SQL, with some functional differences and omissions.
```
Comment (nicecui) [inline docs/en/v0.4/reference/sql/compatibility.md:12]:
```suggestion
4. Query data: Query syntax is compatible with ANSI SQL, with some functional differences and omissions.
```

File: docs/en/v0.4/reference/sql/compatibility.md [hunk 9]
```diff
@@ -1,3 +1,18 @@
+
 # ANSI Compatibility
 
-TODO
+GreptimeDB supports a subset of ANSI SQL and has some unique extensions. Some major incompatibilities and extensions are described below:
+
+1. Table creation:
```
Comment (nicecui) [inline docs/en/v0.4/reference/sql/compatibility.md:6]:
```suggestion
1. Create a table:
```

File: docs/en/v0.4/reference/sql/compatibility.md [hunk 10]
```diff
@@ -1,3 +1,18 @@
...
+   * Supports the unique `TIME INDEX` constraint. Please refer to the [Data Model](/user-guide/concepts/data-model) and the [CREATE](./create.md) table creation syntax for details.
+   * Currently only supports `PRIMARY KEY` constraints and does not support other types of constraints or foreign keys.
+   * GreptimeDB is a native distributed database, so the table creation syntax for distributed tables supports partitioning rules. Please also refer to the [CREATE](./create.md) table creation syntax.
```
Comment (nicecui) [inline docs/en/v0.4/reference/sql/compatibility.md:9]:
```suggestion
   * GreptimeDB is a native distributed database, so the table creation syntax for distributed tables supports partitioning rules. Please also refer to the [CREATE](./create.md).
```

File: docs/en/v0.4/reference/sql/compatibility.md [hunk 11]
```diff
@@ -1,3 +1,18 @@
...
+2. Insertion: Consistent with ANSI SQL syntax, but requires the `TIME INDEX` column value (or default value) to be provided.
+3. Update: Does not support `UPDATE` syntax, but if the primary key and `TIME INDEX` corresponding column values are the same during `INSERT`, subsequent inserted rows will overwrite previously written rows, effectively achieving an update.
+4. Query: Query syntax is compatible with ANSI SQL, with some functional differences and omissions.
+   * Does not support views.
+   * TQL syntax extension: Supports executing PromQL in SQL via TQL subcommands. Please refer to the [TQL](./tql.md) section for details.
+5. Deletion: Deletion syntax is basically consistent with ANSI SQL.
```
Comment (nicecui) [inline docs/en/v0.4/reference/sql/compatibility.md:15]:
```suggestion
5. Delete data: Deletion syntax is basically consistent with ANSI SQL.
```

**Review by nicecui — APPROVED** [review]

**Review by xtang — APPROVED** [review]

**Pull request merged**